File: go/docmost/services/users_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// UsersAPI implements Users API.
type UsersAPI struct {
	http *httpclient.HTTPClient
}

func NewUsersAPI(http *httpclient.HTTPClient) *UsersAPI {
	return &UsersAPI{http: http}
}

func (a *UsersAPI) GetUserInfo(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "users/me", nil)
}

func (a *UsersAPI) UpdateUser(ctx context.Context, request *models.UpdateUserDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "users/update", request)
}

File: python/docmost/services/users_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import UpdateUserDto


class UsersApi:
    """Users API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def get_user_info(self) -> ApiResponse[Any]:
        return self._http.post("users/me", body=None)

    def update_user(self, request: UpdateUserDto) -> ApiResponse[Any]:
        return self._http.post("users/update", body=request.model_dump(by_alias=True, exclude_none=True))

